File: app/src/main/java/com/example/fruitsdiary/usecase/home/HomeAbstractFragment.java
package com.example.fruitsdiary.usecase.home;

import android.support.design.widget.FloatingActionButton;
import android.support.v4.app.Fragment;

public abstract class HomeAbstractFragment extends Fragment {

    public abstract String getFragmentName();

    public abstract void setFabAction(FloatingActionButton fab);
}

File: app/src/main/java/com/example/fruitsdiary/usecase/home/HomeAbstractFragment.kt
package com.example.fruitsdiary.usecase.home

import androidx.fragment.app.Fragment
import com.google.android.material.floatingactionbutton.FloatingActionButton


abstract class HomeAbstractFragment : Fragment() {

    abstract fun  getFragmentName(): String

    abstract fun setFabAction(fab: FloatingActionButton)
}
